### PR TITLE
Jonathan/caseflow 3972 refactor of pre docket task

### PR DESCRIPTION
**Pull request by jtsangVA**
Resolves [CASEFLOW-3972](https://vajira.max.gov/browse/CASEFLOW-3972)

### Description
Refactored PreDocketTask to allow the addition of the Education business line

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] The PreDocketTask is refactored so that it may be utilized for the Education business line
- [ ] Preexisting functionality for the VHA business line is not negatively impacted in any way

### Testing Plan
Run rspec tests:
image

image

1. In appeal.rb, uncomment the elsif block in create_tasks_on_intake_success! and temporarily remove the "ri.is_predocket_needed " check from edu_predocket_needed? if those changes have not been merged yet.
2. In Caseflow, switch to a user with Intake permissions.
3. Navigate to /intake and begin the 10182 process.
4. When you get to the Add/Remove Issues page, add an issue and select "education" as the benefit type.
5. Click "Establish Appeal".
6. Verify that the appeal went into pre-docket queue.
7. Search for the appeal in Caseflow Queue, verify that it is assigned to "Education" and the status is "Pre Docketed", then navigate to the case details page.
8. On the Case Details page, click on the Actions dropdown menu and confirm that "Return Appeal to Education Service" appears.
9. Grab the uuid from the url.
10. Open the rails console and get the appeal's data using the uuid. Verify that the task tree contains the EducationDocumentSearchTask.

image

### User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---
image | image |
image | image

### CODE REVIEW

File: client/app/queue/QueueApp.jsx [hunk 1]
```diff
@@ -344,13 +344,15 @@ class QueueApp extends React.PureComponent {
...
   );
 
-  routedBvaIntakeReturnToCamo = (props) => (
-    <AssignToView isTeamAssign assigneeAlreadySelected {...props.match.params} />
+  routedBvaIntakeReturnToOrg = (props) => (
+    <AssignToView
+      isTeamAssign
+      assigneeAlreadySelected
+      {...props.match.params}
+    />
```
Comment (ajspotts) [inline client/app/queue/QueueApp.jsx:352]: @jtsangVA is this the only actual change in this file i.e. is everything else formatting changes?

Comment (jtsangVA) [reply client/app/queue/QueueApp.jsx:352]: @ajspotts Pretty much. I renamed the function to "routedBvaIntakeReturnToOrg" for reusability and added a route for the Return to Emo option.

File: app/models/tasks/pre_docket/education_document_search_task.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-# Task tracking work searching for decision documents related to VHA issues.
+# Task tracking work searching for decision documents related to Education issues.
```
Comment (ajspotts) [inline app/models/tasks/pre_docket/education_document_search_task.rb:3]: Good catch here!

**Review by ajspotts — APPROVED** [review]
lgtm!

**Pull request merged**